lib/membrane_rtmp_plugin/rtmp_server/listener.ex

```diff
@@ -5,47 +5,120 @@ defmodule Membrane.RTMPServer.Listener do
 
   use Task
   require Logger
-  alias Membrane.RTMPServer.ClientHandler
+  alias Membrane.RTMPServer.{ClientHandler, Config}
 
   @spec run(
           options :: %{
             use_ssl?: boolean(),
             socket_module: :gen_tcp | :ssl,
             server: pid(),
-            port: non_neg_integer()
+            port: non_neg_integer(),
+            ssl_options: keyword()
           }
         ) :: no_return()
   def run(options) do
     options = Map.merge(options, %{socket_module: if(options.use_ssl?, do: :ssl, else: :gen_tcp)})
 
     listen_options =
       if options.use_ssl? do
-        certfile = System.get_env("CERT_PATH")
-        keyfile = System.get_env("CERT_KEY_PATH")
-
-        [
-          :binary,
-          packet: :raw,
-          active: false,
-          certfile: certfile,
-          keyfile: keyfile
-        ]
+        ssl_opts = Config.get_ssl_listen_options(Map.get(options, :ssl_options, []), true)
+
+        Logger.debug("SSL options for listen: #{inspect(ssl_opts)}")
+
+        # SSL listen requires at least certfile and keyfile
+        unless ssl_opts[:certfile] && ssl_opts[:keyfile] do
+          raise ArgumentError, """
+          SSL is enabled but certificate files are not configured.
+          Please configure SSL certificate files via:
+
+          1. Application config:
+             config :membrane_rtmp_plugin, :ssl,
+               certfile: "/path/to/cert.pem",
+               keyfile: "/path/to/key.pem"
+
+          2. Environment variables:
+             RTMP_SSL_CERTFILE="/path/to/cert.pem"
+             RTMP_SSL_KEYFILE="/path/to/key.pem"
+
+          3. Runtime options:
+             ssl_options: [
+               certfile: "/path/to/cert.pem",
+               keyfile: "/path/to/key.pem"
+             ]
+          """
+        end
+
+        # Additional validation for certificate files
+        if ssl_opts[:certfile] && !File.exists?(ssl_opts[:certfile]) do
+          raise ArgumentError, "SSL certificate file does not exist: #{ssl_opts[:certfile]}"
+        end
+
+        if ssl_opts[:keyfile] && !File.exists?(ssl_opts[:keyfile]) do
+          raise ArgumentError, "SSL key file does not exist: #{ssl_opts[:keyfile]}"
+        end
```

Comment:
We already check that `ssl_opts is non `nil` (and otherwise we `raise`) so I don't think we need that check here as well